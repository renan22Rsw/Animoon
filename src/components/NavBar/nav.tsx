interface IMenu {
  id: number;
  label: string;
  href: string;
}

export const Menu: IMenu[] = [
  {
    id: 1,
    label: "Home",
    href: "/",
  },
  {
    id: 2,
    label: "Animes",
    href: "/search/animes",
  },
  {
    id: 3,
    label: "Mangas",
    href: "/search/mangas",
  },
  {
    id: 4,
    label: "Characters",
    href: "/search/characters",
  },
];
