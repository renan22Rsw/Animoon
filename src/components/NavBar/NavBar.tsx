"use client";
import Image from "next/image";

import { Menu } from "./nav";

import logo from "../../app/favicon.ico";

import Link from "next/link";
import MoblieNav from "../SpeedDial/SpeedDial";

const NavBar = () => {
  return (
    <>
      <nav className="bg-[#161616] text-white py-6 px-4 lg:px-32 flex items-center  justify-center rounded-sm">
        <Image src={logo} width={50} height={50} alt="logo" />
        <h1 className="text-white font-bold mx-2 text-xl">AniMoon</h1>

        <div className=" w-full hidden  justify-center lg:flex ">
          <ul className="flex">
            {Menu.map((links) => (
              <li
                className="px-4 text-lg cursor-pointer duration-300 hover:scale-110 ease-in-out"
                key={links.id}
              >
                <Link href={links.href}>{links.label}</Link>
              </li>
            ))}
          </ul>
        </div>
        <MoblieNav />
      </nav>
    </>
  );
};

export default NavBar;
