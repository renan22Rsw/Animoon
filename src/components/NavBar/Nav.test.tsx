import { render, screen } from "@testing-library/react";
import NavBar from "./NavBar";

jest.mock("./nav", () => ({
  Menu: [
    { id: 1, label: "Home", href: "/" },
    { id: 2, label: "Animes", href: "/search/animes" },
    { id: 3, label: "Mangas", href: "/search/mangas" },
    { id: 4, label: "Characters", href: "/search/characters" },
  ],
}));

describe("NavBar component", () => {
  it("should render the logo", () => {
    render(<NavBar />);
    const logoImage = screen.getByRole("img");
    expect(logoImage).toBeInTheDocument();
    expect(logoImage).toHaveAttribute(
      "src",
      expect.stringMatching(/\/_next\/image\?url=%2Fimg.jpg&w=128&q=75/i)
    );
  });

  it("should render menu items", () => {
    render(<NavBar />);

    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByText("Animes")).toBeInTheDocument();
    expect(screen.getByText("Mangas")).toBeInTheDocument();
    expect(screen.getByText("Characters")).toBeInTheDocument();
  });

  it("should render links", () => {
    render(<NavBar />);
    expect(screen.getByText("Home")).toHaveAttribute("href", "/");
    expect(screen.getByText("Animes")).toHaveAttribute(
      "href",
      "/search/animes"
    );
    expect(screen.getByText("Mangas")).toHaveAttribute(
      "href",
      "/search/mangas"
    );
    expect(screen.getByText("Characters")).toHaveAttribute(
      "href",
      "/search/characters"
    );
  });

  it("should render NavMoblie", () => {
    render(<NavBar />);
    expect(screen.getByTestId("moblie-nav")).toBeInTheDocument();
  });

  it("should match a snapshot", () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});
