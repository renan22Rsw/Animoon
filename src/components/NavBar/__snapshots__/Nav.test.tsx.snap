// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should match a snapshot 1`] = `
<div>
  <nav
    class="bg-[#161616] text-white py-6 px-4 lg:px-32 flex items-center  justify-center rounded-sm"
  >
    <img
      alt="logo"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
    <h1
      class="text-white font-bold mx-2 text-xl"
    >
      AniMoon
    </h1>
    <div
      class=" w-full hidden  justify-center lg:flex "
    >
      <ul
        class="flex"
      >
        <li
          class="px-4 text-lg cursor-pointer duration-300 hover:scale-110 ease-in-out"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="px-4 text-lg cursor-pointer duration-300 hover:scale-110 ease-in-out"
        >
          <a
            href="/search/animes"
          >
            Animes
          </a>
        </li>
        <li
          class="px-4 text-lg cursor-pointer duration-300 hover:scale-110 ease-in-out"
        >
          <a
            href="/search/mangas"
          >
            Mangas
          </a>
        </li>
        <li
          class="px-4 text-lg cursor-pointer duration-300 hover:scale-110 ease-in-out"
        >
          <a
            href="/search/characters"
          >
            Characters
          </a>
        </li>
      </ul>
    </div>
    <div
      data-testid="moblie-nav"
    >
      <div
        class="MuiSpeedDial-root MuiSpeedDial-directionUp lg:hidden md:p-10 css-1gu4ml5-MuiSpeedDial-root"
        role="presentation"
      >
        <button
          aria-controls="SpeedDialopenIconexample-actions"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="SpeedDial openIcon example"
          class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiSpeedDial-fab css-7dv1rb-MuiButtonBase-root-MuiFab-root-MuiSpeedDial-fab"
          style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiSpeedDialIcon-root css-ohncnb-MuiSpeedDialIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSpeedDialIcon-openIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSpeedDialIcon-icon css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          aria-orientation="vertical"
          class="MuiSpeedDial-actions MuiSpeedDial-actionsClosed css-1s28u9o-MuiSpeedDial-actions"
          id="SpeedDialopenIconexample-actions"
          role="menu"
        >
          <button
            aria-label="Home"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-13srr63-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
            data-mui-internal-clone-element="true"
            role="menuitem"
            style="transition-delay: 120ms;"
            tabindex="-1"
            type="button"
          >
            <a
              href="/"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Search"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-13srr63-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
            data-mui-internal-clone-element="true"
            role="menuitem"
            style="transition-delay: 90ms;"
            tabindex="-1"
            type="button"
          >
            <a
              href="/search/animes"
            >
               
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                />
              </svg>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Mangas"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-13srr63-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
            data-mui-internal-clone-element="true"
            role="menuitem"
            style="transition-delay: 60ms;"
            tabindex="-1"
            type="button"
          >
            <a
              href="/search/mangas"
            >
               
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuBookIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 5c-1.11-.35-2.33-.5-3.5-.5-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5V6c-.6-.45-1.25-.75-2-1m0 13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5z"
                />
                <path
                  d="M17.5 10.5c.88 0 1.73.09 2.5.26V9.24c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99M13 12.49v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26V11.9c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.3-4.5.83m4.5 1.84c-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26v-1.52c-.79-.16-1.64-.24-2.5-.24"
                />
              </svg>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Characters"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-13srr63-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
            data-mui-internal-clone-element="true"
            role="menuitem"
            style="transition-delay: 30ms;"
            tabindex="-1"
            type="button"
          >
            <a
              href="/search/characters"
            >
               
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4m0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4"
                />
              </svg>
              ,
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
